swagger.yaml

```diff
@@ -3153,6 +3153,48 @@ paths:
             $ref: '#/definitions/error'
 
 definitions:
+  baremetal_platform:
+    description: Baremetal platform specific configuration upon which to perform the installation
+    type: object
+    properties:
+      provisioningNetwork:
+        type: string
+        enum: [Disabled, Enabled]
+
+  vsphere_platform:
+    type: object
+    description: Vsphere platform specific configuration upon which to perform the installation
+    required:
+      - username
```

Comment:
why required? didn't we say it is not must?

Reply:
Required for day2 operations to work

Reply:
I added the `type` under platform, if type is set and no details (`vsphere_platform`) is given it will act as vsphere platform without credentials. Those fields are mandatory when the `vsphere` platform is selected and the `vsphere_platform` is initialized.

Reply:
If it can be installed successfully without passing any parameters then this is the best way to go (either pass everything or nothing).

Reply:
@avishayt yep, but i think it will be very helpful to allow to provide all params except user/password
This is will allow to make day2 operation much easier

Reply:
@tsorya Why?  What's the difference?  It's possible to update the user/password but not the other information?

Reply:
User password is copied to secret and all other info should be updated in some other place
@eliorerz can you please check where we should update vspere details? 
My point it that if we can minimize user day2 actions by given him to set all params without credentials it will allow us to reduce his day2 operations and make process more friendly

Reply:
In any case we will need to push some junk data there as it is required by installer? @eliorerz ?

Reply:
Is the vmware team OK with this?  This seems kind of hacky.

Reply:
@eliorerz ?

Reply:
We decided not to put placeholders at first and so the `install_config.yaml` platform (if properties are not set) will look something like this:
```
paltform:
  vsphere:
    apiVIP: 2.2.2.2
    ingressVIP: 1.1.1.1
```